### PR TITLE
feature: add a contract that supports voting on economic contracts

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
refs #4142

## Description

add a contract that supports voting on economic contracts

### Security Considerations

...lots to say here...

### Documentation Considerations

TBD

### Testing Considerations

TBD

### CODE REVIEW

File: packages/run-protocol/src/voting.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    deadline,
+  ) => {
+    return E(target).voteOnParamChanges(counter, deadline, {
+      paramPath: path,
+      changes: params,
+    });
+  };
+
+  const voteOnApiInvocation = (contract, method, amounts, deadline) => {
+    return E(reserve).voteOnApiInvocation(method, amounts, counter, deadline);
```
Comment (dckc) [inline packages/run-protocol/src/voting.js]:
```suggestion
    return E(contract).voteOnApiInvocation(method, amounts, counter, deadline);
```

Comment (Chris-Hibbert) [reply packages/run-protocol/src/voting.js]: That was intentional, since reserve is the only one with an ApiInvocation, but we can make it general.

Comment (dckc) [reply packages/run-protocol/src/voting.js]: or take out the `contract` arg

Comment (Chris-Hibbert) [reply packages/run-protocol/src/voting.js]: Aah.  you're right. I was part way through a conversion. I think making it general was the right direction to go.

**Comment by dckc** [comment]
We incorporated this into #5257 ... I'm not sure whether it's worth keeping this open.

**Comment by michaelfig** [comment]
> I'm not sure whether it's worth keeping this open.

I'll express my opinion via the Close button.

**Pull request closed without merging**